### PR TITLE
Merge templates

### PR DESCRIPTION
**Pull request by david-i-berry**
Merge of csv2bufr_templates and csv2bufr modules.
- Templates can now be loaded with csv2bufr.templates.load_template()
- Default templates provided
- User defined path for templates added (CSV2BUFR_TEMPLATES=...)

### CODE REVIEW

**Comment by david-i-berry** [comment]
@tomkralidis @maaikelimper I've refactored where the default templates are in the module. I don't know if these are used in the wis2box but something to be aware off when reviewing.

File: csv2bufr/templates/__init__.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+
+def load_template(template_name):
+    template = None
+    # iterate over directories and load file
+    for dir in TEMPLATE_DIRS:
+        try:
+            template_file = f"{dir}{os.sep}{template_name}.json"
+            if os.path.isfile(template_file):
+                with open(template_file, 'r') as fh:
```
Comment (tomkralidis) [inline csv2bufr/templates/__init__.py:47]:
```python
with template_file.open() as fh:
```

File: csv2bufr/templates/__init__.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    if template is None:
+        LOGGER.warning(f"Requested template '{template_name}' not found." +
+                       f" Search path = {TEMPLATE_DIRS}")
+
+    return template
+
+
+def list_templates():
+    templates = []
+    for dir in TEMPLATE_DIRS:
```
Comment (tomkralidis) [inline csv2bufr/templates/__init__.py:62]: `dir_`

File: csv2bufr/templates/__init__.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        try:
+            _templates = os.listdir(dir)
```
Comment (tomkralidis) [inline csv2bufr/templates/__init__.py:64]: Suggest to use pathlib's `iterdir` generator.

```python
for template in dir_.iterdir():
```

File: csv2bufr/templates/__init__.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+# Set user defined location first
+if 'CSV2BUFR_TEMPLATES' in os.environ:
+    TEMPLATE_DIRS.append(os.environ['CSV2BUFR_TEMPLATES'])
+
+# Now add defaults
+TEMPLATE_DIRS.append(
+    f"{os.path.dirname(os.path.realpath(__file__))}{os.sep}resources")
+
+
+def load_template(template_name):
```
Comment (tomkralidis) [inline csv2bufr/templates/__init__.py:40]: Add docstring and function annotations.

File: csv2bufr/templates/__init__.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            LOGGER.warning(f"Error raised loading csv2bufr templates: {e}.")
+
+    if template is None:
+        LOGGER.warning(f"Requested template '{template_name}' not found." +
+                       f" Search path = {TEMPLATE_DIRS}")
+
+    return template
+
+
+def list_templates():
```
Comment (tomkralidis) [inline csv2bufr/templates/__init__.py:60]: Add docstring and function annotations.

**Review by tomkralidis — CHANGES_REQUESTED** [review]

**Comment by tomkralidis** [comment]
Thanks @david-i-berry. To confirm, https://github.com/wmo-im/csv2bufr-templates is now not to be used?  Should this repo be set to archived mode then?

**Comment by david-i-berry** [comment]
> Thanks @david-i-berry. To confirm, https://github.com/wmo-im/csv2bufr-templates is now not to be used? Should this repo be set to archived mode then?

Yes, I'll mark it to archived now and add a note.

**Review by tomkralidis — APPROVED** [review]

**Pull request merged**